tiledb/sm/query/readers/attribute_order_validator.h

```diff
@@ -0,0 +1,581 @@
+      if (!min_validated_[f]) {
+        // See if the min is covered by this fragment.
+        min_validated_[f] |= in_domain(min, non_empty_domain2);
+
+        // If the min is next to the max of a more recent fragment, it will
+        // be validated when processing that fragment.
+        min_validated_[f] |= min - 1 == non_empty_domain2[1];
+      }
+
+      // Check if the upper bound can be validated.
+      if (!max_validated_[f]) {
+        // See if the max is covered.
+        max_validated_[f] |= in_domain(max, non_empty_domain2);
+
+        // If the max is next to the min of a more recent fragment, it will
+        // be validated when processing that fragment.
+        max_validated_[f] |= max + 1 == non_empty_domain2[0];
+      }
+
+      // If both bounds are validated, then no fragments need to be checked and
+      // we can return.
+      if (min_validated_[f] && max_validated_[f]) {
+        return;
+      }
+    }
+
+    // Get fragment to check against for both the lower and upper boundaries
+    // of this fragment.
+    bool finished_lower_search = min_validated_[f];
+    bool finished_upper_search = max_validated_[f];
+    for (int64_t f2 = f - 1; f2 >= 0; --f2) {
+      // Get the non-empty domain for this fragment.
+      const IndexType* non_empty_domain2 =
+          static_cast<const IndexType*>(non_empty_domains[f2]);
+
+      // If not validated and fragment to check is not yet found, check if
+      // this fragment is over-lapping or directly preceeding the min.
+      if (!finished_lower_search && (in_domain(min, non_empty_domain2) ||
+                                     min - 1 == non_empty_domain2[1])) {
+        frag_to_compare.first = f2;
+        finished_lower_search = true;
+      }
+
+      // If not validated and fragment to check is not yet found, check if
+      // this fragment is over-lapping or directly following the max.
+      if (!finished_upper_search && (in_domain(max, non_empty_domain2) ||
+                                     max + 1 == non_empty_domain2[0])) {
+        frag_to_compare.second = f2;
+        finished_upper_search = true;
+      }
+
+      // If both fragments are either validated or the matching fragment was
+      // found, then we are done searching.
+      if (finished_lower_search && finished_upper_search) {
+        return;
+      }
+    }
+
+    // If the search/validation failed, then there is  a discontinuity in this
+    // array.
+    throw AttributeOrderValidatorStatusException(
+        "Discontiuity found in array domain");
+  }
+
+  /**
+   * Performs validation that can be done without loading tile.
+   *
+   * If a validation check fails, this will throw an error.
+   *
+   * @tparam Index type.
+   * @tparam Attribute type.
+   * @param index_dim The index dimension.
+   * @param f Fragment index.
+   * @param non_empty_domains Vector of pointers to the non-empty domain for
+   *     each fragment.
+   *
+   */
```

Comment:
```suggestion
  /**
   * Performs validation that can be done without loading tile.
   *
   * If a validation check fails, this will throw an error. This requires
   * ``find_fragments_to_check`` to be ran prior to execution.  
   *
   * @tparam Index type.
   * @tparam Attribute type.
   * @param index_dim The index dimension.
   * @param f Fragment index.
   * @param non_empty_domains Vector of pointers to the non-empty domain for
   *     each fragment.
   */
```